systemtests/src/main/java/io/enmasse/systemtest/manager/SharedIoTManager.java

```diff
@@ -0,0 +1,192 @@
+            var iotProjectApiClient = kubernetes.getIoTProjectClient(sharedIoTProject.getMetadata().getNamespace());
+            if (iotProjectApiClient.withName(sharedIoTProject.getMetadata().getName()).get() != null) {
+                IoTUtils.deleteIoTProjectAndWait(kubernetes, sharedIoTProject);
+                sharedIoTProject = null;
+            } else {
+                LOGGER.info("IoTProject '{}' doesn't exists!", sharedIoTProject.getMetadata().getName());
+            }
+        }
+        tearDownSharedIoTConfig();
+        SystemtestsKubernetesApps.deleteInfinispanServer(kubernetes.getInfraNamespace());
+    }
+
+    public void tearDownSharedIoTConfig() throws Exception {
+        if (sharedIoTConfig != null) {
+            LOGGER.info("Shared IoTConfig will be removed");
+            var iotConfigApiClient = kubernetes.getIoTConfigClient();
+            if (iotConfigApiClient.withName(sharedIoTConfig.getMetadata().getName()).get() != null) {
+                IoTUtils.deleteIoTConfigAndWait(kubernetes, sharedIoTConfig);
+                sharedIoTConfig = null;
+            } else {
+                LOGGER.info("IoTConfig '{}' doesn't exists!", sharedIoTConfig.getMetadata().getName());
+            }
+        }
+    }
+
+    @Override
+    void initFactories(AddressSpace addressSpace) {
+
+        amqpClientFactory = new AmqpClientFactory(getSharedAddressSpace(), defaultCredentials);
+        mqttClientFactory = new MqttClientFactory(getSharedAddressSpace(), defaultCredentials);
+    }
+
+    @Override
+    public void setup() {
+        if (!kubernetes.namespaceExists(iotProjectNamespace)) {
+            LOGGER.info("Namespace {} doesn't exists and will be created.");
+            kubernetes.createNamespace(iotProjectNamespace);
+        }
+    }
+
+    public void createSharedIoTEnv() throws Exception {
+        Environment.getInstance().setDefaultCredentials(new UserCredentials(UUID.randomUUID().toString(), UUID.randomUUID().toString()));
+
+        if (sharedIoTConfig == null) {
+            createNewIoTConfig();
+        }
+
+        if (sharedIoTProject == null) {
+            sharedIoTProject = IoTUtils.getBasicIoTProjectObject("shared-iot-project",
+                    DEFAULT_ADD_SPACE_IDENTIFIER, iotProjectNamespace, ADDRESS_SPACE_PLAN);
+            createIoTProject(sharedIoTProject);
+        }
+        initFactories(getSharedAddressSpace());
+        createOrUpdateUser(getSharedAddressSpace(), defaultCredentials);
+        this.amqpClient = amqpClientFactory.createQueueClient();
+    }
+
+    private void createNewIoTConfig() throws Exception {
+        CertBundle certBundle = CertificateUtils.createCertBundle();
+        sharedIoTConfig = new IoTConfigBuilder()
+                .withNewMetadata()
+                .withName("default")
+                .endMetadata()
+                .withNewSpec()
+                .withNewServices()
+                .withDeviceRegistry(deviceRegistry("file"))
```

Comment:
Shouldn't this default to "infinispan"?

Reply:
I think default device registry was declared by this system env value which said default value is file https://github.com/EnMasseProject/enmasse/blob/master/systemtests/src/main/java/io/enmasse/systemtest/Environment.java#L55

Reply:
So I think the default should be "infinispan", as this is the one we support. "file" is only there for PoCs and stuff that doesn't matter. So the default should really be "infinispan".